packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -0,0 +1,588 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+import '../auto_dispose_mixin.dart';
+import 'chart_controller.dart';
+import 'chart_trace.dart';
+
+class Chart extends StatefulWidget {
+  Chart(this.controller, String title) {
+    controller.title = title;
+  }
+
+  final ChartController controller;
+
+  @override
+  ChartState createState() => ChartState();
+}
+
+class ChartState extends State<Chart> with AutoDisposeMixin {
+  ChartController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = widget.controller;
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    cancel();
+
+    addAutoDisposeListener(controller.traceChanged, () {
+      setState(() {
+        controller.computeChartArea();
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.symmetric(horizontal: 5, vertical: 10),
+      child: LayoutBuilder(
+        // Inner container
+        builder: (_, constraints) => Container(
+          width: constraints.widthConstraints().maxWidth,
+          height: constraints.widthConstraints().maxHeight,
+          color: Colors.black,
```

Comment:
why black? This should be from the theme or we shouldn't specify a background.

Reply:
Right it should be specified (that would be a passed in theme).

Reply:
Fixed